src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+
+import command.ByeCommand;
+import command.FindCommand;
+import command.EventCommand;
+import command.DeleteCommand;
+import command.DeadlineCommand;
+import command.MarkCommand;
+import command.TodoCommand;
+import command.UnmarkCommand;
+import command.ListCommand;
+import command.Command;
+import exceptions.BlankException;
+import exceptions.InvalidInstructionException;
+import exceptions.DukeException;
+
+/**
+ * Handles user input.
+ *
+ * @author Julio Harjo
+ */
+public class Parser {
+
+    /**
+     * Stores tasks in a text file.
+     */
+    protected static Storage storage;
+
+    /**
+     * Handles User interaction.
+     */
+    protected static Ui ui;
+
+    /**
+     * Constructor.
+     *
+     * @param storage The storage list used.
+     * @param ui      The duke.duke.Ui interface used.
+     */
+    public Parser(Storage storage, Ui ui) {
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /***
+     * parse the instruction accordingly based on first word in the string
+     * @param command
+     * @return
+     * @throws DukeException
+     */
+    public static Command parse(String command) throws DukeException {
+        String[] c = command.split(" ", 2);
+        String instruction = c[0];
+        Command com = null;
+        try {
+            switch (instruction) {
+                case ("list"):
+                    com = new ListCommand();
+                    break;
+                case ("find"):
+                    checkIfBlank(c);
```

Comment:
My suggestion is that all the validity checking work can be put inside the corresponding command classes, but no big deal.